File: OJ/Leetcode/Algorithm/121. Best Time to Buy and Sell Stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if prices == []:
            return 0
        maxn = prices[-1]
        ans = 0
        for num in reversed(prices):
            maxn = max(num, maxn)
            ans = max(ans, maxn-num)
        return ans

File: OJ/Leetcode/Algorithm/121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minn = INT_MAX;
        int ans = 0;
        for(auto p : prices)
        {
            if(p < minn)
                minn = p;
            ans = max(ans, p - minn);
        }
        return ans;
    }
};
